main.cpp

```diff
@@ -0,0 +1,228 @@
+        if (temp_value == "") {
+            config_file << "[knownrecipients]=./knownrecipients.txt" << endl;
+        } else {
+            config_file << "[knownrecipients]=" << temp_value << endl;
+        }
+
+        temp_value = "";
+
+        config_file.close();
+
+
+    } else if (argument1 == "edit") {
+        // edit the config file
+
+        // Create a backup of the config file
+        const char * orig_file = CONFIG_FILE_PATH.c_str();
+        const char * backup_file = (CONFIG_FILE_PATH + ".old").c_str();
+
+        // Delete the previous backup if it exists
+        std::remove(backup_file);
+
+        // Create a backup of the config before it is edited
+        int result = std::rename(orig_file, backup_file);
+
+        if (result == 0) {
+
+            // TODO if the edit item is missing then cycle through the
+            // config file allowing the user to accept the current settings
+
+            std::ifstream infile;
+            std::ofstream outfile;
+
+            infile.open(backup_file);
+            outfile.open(orig_file);
+
+            // Check if the second argument is blank
+            string command_argument = "";
+            if (argv[2] != nullptr) {
+                command_argument = static_cast<string>(argv[2]);
+            }
+
+
+            string read_line = "";
+            string current_setting = "";
+            string new_setting = "";
+
+            if (infile.is_open()) {
+                while (std::getline(infile, read_line)) {
+
+                    // Get the setting from the each line
+                    string temp_value = "";
+                    string setting_value = "";
+                    string setting_name = "";
+                    bool found_first_bracket = false; // Make true when the first '[' is found
+
+                    for (int i = 0; i < read_line.length(); i++) {
+                        if (found_first_bracket) {
+                            // Check for the next "]"
+                            if (read_line[i] == ']') {
+
+                                setting_name = temp_value;
+                                setting_value = read_line.substr(i + 2, read_line.length() - i + 2);
+
+                                found_first_bracket = false;
+                                temp_value = "";
+                                break;
+                            } else {
+                                // Enter the line one char at a time and convert the char
+                                // to lower case for comparison.
+                                std::locale loc;
+                                temp_value = temp_value + std::tolower(read_line[i], loc);
+
+                            }
+                        } else {
+                            if (read_line[i] == '[') {
+                                found_first_bracket = true;
+                            }
+                        }
+
+                    }
```

Comment:
I do like how this doesn't rely on the order of the items in the file. That is a very good idea.